src/plugins/intel_gpu/src/plugin/transformations/unsqueeze_broadcast_reshape_matmul_fusion.cpp

```diff
@@ -25,118 +26,63 @@ UnsqueezeBroadcastReshapeMatmulFusion::UnsqueezeBroadcastReshapeMatmulFusion() {
         return std::dynamic_pointer_cast<ov::op::v1::Reshape>(output.get_node_shared_ptr()) == nullptr;
     };
 
-    auto broadcast_rank_equals_and_has_static_dims = [](const ov::Output<ov::Node>& output) -> bool {
-        return rank_equals(5)(output) && has_static_dims({2, 3}) && consumers_count(1);
+    auto unsqueeze_predicate = [](const ov::Output<ov::Node>& output) -> bool {
+        const auto unsqueeze = ov::as_type_ptr<const ov::op::v0::Unsqueeze>(output.get_node_shared_ptr());
+        if (!unsqueeze)
+            return false;
+        const auto kvcache = ov::as_type_ptr<const ov::intel_gpu::op::KVCache>(unsqueeze->get_input_node_shared_ptr(0));
```

Comment:
Why it's checked in predicate? I think KVCache node can be added to the pattern explicitly, can't it?

```cpp
    auto input_b_m = wrap_type<ov::intel_gpu::op::KVCache>({/*inputs*/});
    auto axes_const_m = wrap_type<ov::op::v0::Constant>();
    auto unsqueeze_m = wrap_type<ov::op::v0::Unsqueeze>({input_b_m, axes_const_m}, unsqueeze_predicate);
```

Reply:
yes. It's possible. I confirmed that it's not included in matched nodes list because it's an input. updated as your suggestion.